prediction_market_agent/agents/microchain_agent/memory_functions.py

```diff
@@ -94,6 +97,44 @@ def __call__(self, context: str) -> str:
         return memories_to_learnings(memories=results, model=self.model)
 
 
+def fetch_memories_from_last_run(
+    agent_identifiers: list[AgentIdentifier],
+) -> dict[str, list[LongTermMemories]]:
+    """
+    Returns memories from the last run for each agent.
+    The last run is identified by looking at "GameRoundEnd" function calls
+    as part of the prompt history of each agent.
+    """
+    entries_from_latest_run: dict[str, list[LongTermMemories]] = {}
+    for agent_id in agent_identifiers:
+        ltm = LongTermMemoryTableHandler.from_agent_identifier(agent_id)
+        entries_for_agent = ltm.search()
+        # We need the 2nd GameRoundEnd. If there is only one, then start_date should be None.
+        game_round_occurrences = [
+            i
+            for i in entries_for_agent
+            if i.metadata_dict is not None and "content" in i.metadata_dict
+            and "GameRoundEnd(" in i.metadata_dict["content"]
```

Comment:
```suggestion
            and f"{GameRoundEnd.__name__}(" in i.metadata_dict["content"]
```

This way, if we ever change `GameRoundEnd`'s name or remove it entirely, mypy will complain, and we know this needs to be updated accordingly.